File: src/minigames/pillar_battle/java/work/lclpnet/ap2/game/pillar_battle/item/ItemClass.java
package work.lclpnet.ap2.game.pillar_battle.item;

import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;

import java.util.Random;
import java.util.stream.Stream;

public interface ItemClass {

    ItemStack getRandomStack(Random random);

    Stream<Item> stream();
}

File: src/minigames/pillar_battle/src/main/kotlin/work/lclpnet/ap2/game/pillar_battle/item/ItemClass.kt
package work.lclpnet.ap2.game.pillar_battle.item

import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import java.util.*
import java.util.stream.Stream

interface ItemClass {
    fun getRandomStack(random: Random): ItemStack
    fun stream(): Stream<Item>
}
